### PR TITLE
[Api][Web][SIMS #1125] Reorganize institution API part 2 final

### PR DESCRIPTION
**Pull request by dheepak-aot**
- [x] The APIs controller institution.controller.ts is moved to the respective client type controllers.
- [x] DTOs in API side and Vue side have been refactored as per the naming convention.
- [x] Old code with Pagination API(APIs in the scope of story) refactored to use pagination utils developed later.

API URL for Institution search modified to avoid the route conflict. @Get("all/search")

### CODE REVIEW

File: sources/packages/api/src/route-controllers/institution/institution.aest.controller.ts [hunk 1]
```diff
@@ -64,4 +79,96 @@ export class InstitutionAESTController extends BaseController {
...
+    );
+  }
+
+  /**
+   * Search the institution based on the search criteria.
+   * @param legalName legalName of the institution.
+   * @param operatingName operatingName of the institution.
+   * @returns Searched institution details.
+   */
+  @Get("all/search")
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/route-controllers/institution/institution.aest.controller.ts:123]: Is the "all" added here to avoid route conflicts? Is it part of the refactor or was it already there?
Do we really need it?

File: sources/packages/api/src/route-controllers/institution/institution.aest.controller.ts [hunk 2]
```diff
@@ -64,4 +79,96 @@ export class InstitutionAESTController extends BaseController {
...
+        city: eachInstitution.institutionAddress.city,
+        provinceState: eachInstitution.institutionAddress.provinceState,
+        country: eachInstitution.institutionAddress.country,
+        postalCode: eachInstitution.institutionAddress.postalCode,
+      },
+    }));
+  }
+
+  /**
+   * Get the Basic Institution info for the ministry institution detail page
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/route-controllers/institution/institution.aest.controller.ts:153]: Please add period.

File: sources/packages/api/src/route-controllers/institution/institution.aest.controller.ts (hunk 1 above)
Comment (dheepak-aot) [reply sources/packages/api/src/route-controllers/institution/institution.aest.controller.ts:123]: Yes it was introduced to avoid route conflicts. This was conflicting with @Get(":institutionId") in the same controller.
Let me add that to PR description.

File: sources/packages/api/src/route-controllers/institution/institution.institutions.controller.ts [hunk 3]
```diff
@@ -27,10 +69,77 @@ export class InstitutionInstitutionsController extends BaseController {
...
+    @Query(PaginationParams.SearchCriteria) searchName: string,
+    @Query(PaginationParams.SortField) sortField: string,
+    @UserToken() user: IInstitutionUserToken,
+    @Query(PaginationParams.SortOrder) sortOrder = FieldSortOrder.ASC,
+    @Query(PaginationParams.Page) page = DEFAULT_PAGE_NUMBER,
+    @Query(PaginationParams.PageLimit) pageLimit = DEFAULT_PAGE_LIMIT,
+  ): Promise<PaginatedResults<InstitutionUserAPIOutDTO>> {
+    return this.institutionControllerService.getInstitutionUsers(
+      user.authorizations.institutionId,
+      {
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/route-controllers/institution/institution.institutions.controller.ts:133]: if we renamed _searchName_ to _searchCriteria_ the entire object can be simplified as below.
```typescript
{ page, pageLimit, searchCriteria, sortField, sortOrder}
```

File: sources/packages/api/src/route-controllers/institution/institution.institutions.controller.ts [hunk 4]
```diff
@@ -52,12 +161,358 @@ export class InstitutionInstitutionsController extends BaseController {
...
+
+  /**
+   * Create an Institution user.
+   * @param payload
+   * @param user
+   * @returns Primary identifier of the created resource.
+   */
+  @ApiUnprocessableEntityResponse({
+    description:
+      "User to be added either not found in account service or does not belong to same institution of logged in user.",
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/route-controllers/institution/institution.institutions.controller.ts:181]: Can we mention here _BCeID account service_ instead of only _account service_? It could help someone looking from outside to figure it out easier.

File: sources/packages/api/src/route-controllers/institution/institution.institutions.controller.ts [hunk 5]
```diff
@@ -52,12 +161,358 @@ export class InstitutionInstitutionsController extends BaseController {
...
+
+      if (legalSigningAuthority) {
+        throw new UnprocessableEntityException(
+          LEGAL_SIGNING_AUTHORITY_EXIST,
+          LEGAL_SIGNING_AUTHORITY_MSG,
+        );
+      }
+    }
+
+    // Create the user user and the related records.
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/route-controllers/institution/institution.institutions.controller.ts:232]: Please adjust the comment to remove the duplicated "user user".

File: sources/packages/api/src/route-controllers/institution/institution.institutions.controller.ts (hunk 4 above)
Comment (dheepak-aot) [reply sources/packages/api/src/route-controllers/institution/institution.institutions.controller.ts:181]: I had the same feeling while moving the code 😊. Can change it.

File: sources/packages/api/src/route-controllers/institution/institution.institutions.controller.ts [hunk 6]
```diff
@@ -52,12 +161,358 @@ export class InstitutionInstitutionsController extends BaseController {
...
+  @ApiUnprocessableEntityResponse({
+    description: "User not active.",
+  })
+  @IsInstitutionAdmin()
+  @Get("user/:userName")
+  async getInstitutionUserByUserName(
+    @Param("userName") userName: string,
+  ): Promise<InstitutionUserAPIOutDTO> {
+    // Get institutionUser
+    const institutionUser =
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/route-controllers/institution/institution.institutions.controller.ts:270]: It seems that there is an authorization check missing here. How can we ensure that the user has access to the _userName_ requested? If not part of the PR please add a TODO here.

Comment (dheepak-aot) [reply sources/packages/api/src/route-controllers/institution/institution.institutions.controller.ts:270]: What is the missing authorization check ?

File: sources/packages/api/src/route-controllers/institution/institution.institutions.controller.ts [hunk 7]
```diff
@@ -52,12 +161,358 @@ export class InstitutionInstitutionsController extends BaseController {
...
+  @ApiNotFoundResponse({
+    description: "User to be updated not found.",
+  })
+  @ApiUnprocessableEntityResponse({
+    description:
+      "User to be updated doesn't belong to institution of logged in user.",
+  })
+  @IsInstitutionAdmin()
+  @Patch("user/:userName")
+  async UpdateInstitutionUserWithAuth(
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/route-controllers/institution/institution.institutions.controller.ts:318]: Please change the method to start with a lower case.

File: sources/packages/api/src/route-controllers/institution/institution.institutions.controller.ts [hunk 8]
```diff
@@ -52,12 +161,358 @@ export class InstitutionInstitutionsController extends BaseController {
...
+    // Check its a active user
+    const institutionUser =
+      await this.institutionService.getInstitutionUserByUserName(userName);
+
+    if (!institutionUser) {
+      throw new NotFoundException("User to be updated not found");
+    }
+
+    // checking if user belong to logged-in users institution
+    if (institutionUser.institution.id !== token.authorizations.institutionId) {
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/route-controllers/institution/institution.institutions.controller.ts:332]: I believe that this is the authorization check that is missing in the above method `getInstitutionUserByUserName`.

File: sources/packages/api/src/route-controllers/institution/institution.institutions.controller.ts [hunk 9]
```diff
@@ -52,12 +161,358 @@ export class InstitutionInstitutionsController extends BaseController {
...
+   * @param userToken
+   * @returns Location details.
+   */
+  @Get("my-locations")
+  async getMyInstitutionLocations(
+    @UserToken() userToken: IInstitutionUserToken,
+  ): Promise<InstitutionUserLocationsAPIOutDTO[]> {
+    // get all institution locations that user has access too.
+    const InstitutionLocationData = userToken.authorizations.isAdmin()
+      ? await this.locationService.getAllInstitutionLocations(
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/route-controllers/institution/institution.institutions.controller.ts:459]: I believe that we no longer need this condition. In the past `.getLocationsIds()` was not loaded for admins users, but they are now supposed to be present due to the authorization. Hence the call `getMyInstitutionLocations` should work for both types of users.
If not part of the PR please add a TODO.

File: sources/packages/api/src/route-controllers/institution/institution.aest.controller.ts [hunk 10]
```diff
@@ -38,10 +53,10 @@ export class InstitutionAESTController extends BaseController {
    * @param institutionId
    * @returns InstitutionDetailDTO.
    */
-  @Get("/:institutionId")
```
Comment (ann-aot) [inline sources/packages/api/src/route-controllers/institution/institution.aest.controller.ts:41]: oh.. will this also work?

**Review by andrewsignori-aot — COMMENTED** [review]
Great work and thanks for the additional refactors done along the way.
Please take a look at the comments.

File: sources/packages/api/src/route-controllers/institution/institution.aest.controller.ts (hunk 10 above)
Comment (dheepak-aot) [reply sources/packages/api/src/route-controllers/institution/institution.aest.controller.ts:41]: Yes it works. I assume you are asking about the leading slash. Btw. It got carried from old code.

File: sources/packages/api/src/route-controllers/institution/institution.institutions.controller.ts (hunk 6 above)
Comment (dheepak-aot) [reply sources/packages/api/src/route-controllers/institution/institution.institutions.controller.ts:270]: I got the point after seeing the comment below.

File: sources/packages/api/src/route-controllers/institution/institution.aest.controller.ts (hunk 1 above)
Comment (andrewsignori-aot) [reply sources/packages/api/src/route-controllers/institution/institution.aest.controller.ts:123]: There is a way to order the methods to avoid it. If I am not wrong @ann-aot did it in another PR to resolve the same scenario.

File: sources/packages/api/src/route-controllers/institution/institution.institutions.controller.ts (hunk 3 above)
Comment (andrewsignori-aot) [reply sources/packages/api/src/route-controllers/institution/institution.institutions.controller.ts:135]: I saw that you gave a thumbs up for this. Was it changed?

Comment (dheepak-aot) [reply sources/packages/api/src/route-controllers/institution/institution.institutions.controller.ts:135]: Please refresh and see again.

File: sources/packages/api/src/route-controllers/institution/institution.aest.controller.ts [hunk 11]
```diff
@@ -64,4 +79,96 @@ export class InstitutionAESTController extends BaseController {
...
+    @Query(PaginationParams.SearchCriteria) searchName: string,
+    @Query(PaginationParams.SortField) sortField: string,
+    @Query(PaginationParams.SortOrder) sortOrder = FieldSortOrder.ASC,
+    @Query(PaginationParams.Page) page = DEFAULT_PAGE_NUMBER,
+    @Query(PaginationParams.PageLimit) pageLimit = DEFAULT_PAGE_LIMIT,
+  ): Promise<PaginatedResults<InstitutionUserAPIOutDTO>> {
+    return this.institutionControllerService.getInstitutionUsers(
+      institutionId,
+      {
+        page: page,
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/route-controllers/institution/institution.aest.controller.ts:108]: This one can also be simplified, not a blocker.

**Review by andrewsignori-aot — APPROVED** [review]
Looks good 👍 
Great refactors! Thanks for thanking the opportunity and for improving the old code also. ❤️

File: sources/packages/api/src/services/institution/institution.service.e2e-spec.ts [hunk 12]
```diff
@@ -84,12 +85,9 @@ describe("InstitutionService", () => {
     await service.remove(user);
   });
 
-  it("should return all institution users", async () => {
+  it.skip("should return all institution users", async () => {
```
Comment (guru-aot) [inline sources/packages/api/src/services/institution/institution.service.e2e-spec.ts:88]: Why did we skip this test?

Comment (dheepak-aot) [reply sources/packages/api/src/services/institution/institution.service.e2e-spec.ts:88]: It looked like we need to invest some time to get these test cases fixed. And also service which was part of the test is working fine. so test was skipped.

Comment (dheepak-aot) [reply sources/packages/api/src/services/institution/institution.service.e2e-spec.ts:88]: Btw, it is intentional 😊

File: sources/packages/api/src/route-controllers/institution/institution.aest.controller.ts (hunk 10 above)
Comment (ann-aot) [reply sources/packages/api/src/route-controllers/institution/institution.aest.controller.ts:41]: ya

File: sources/packages/api/src/route-controllers/institution/institution.aest.controller.ts [hunk 13]
```diff
@@ -64,4 +79,96 @@ export class InstitutionAESTController extends BaseController {
...
+   * passed then DEFAULT_PAGE_LIMIT is taken
+   * @queryParm searchName, user's name keyword to be searched
+   * @queryParm sortField, field to be sorted
+   * @queryParm sortOrder, order to be sorted
+   * @returns All the institution users for the given institution
+   * with total count.
+   */
+  @Get(":institutionId/user")
+  async getInstitutionUsers(
+    @Param("institutionId") institutionId: number,
```
Comment (ann-aot) [inline sources/packages/api/src/route-controllers/institution/institution.aest.controller.ts:98]: please add the param in the comment

File: sources/packages/api/src/route-controllers/institution/institution.aest.controller.ts (hunk 1 above)
Comment (ann-aot) [reply sources/packages/api/src/route-controllers/institution/institution.aest.controller.ts:123]: ya , you can this is controller before the `@Get(":institutionId")` controller

File: sources/packages/web/src/components/common/InstitutionUserSummary.vue [hunk 14]
```diff
@@ -213,29 +213,17 @@ export default {
...
-              searchBox.value,
-              sortField,
-              sortOrder,
-            );
-          break;
-      }
+      usersListAndCount.value =
+        await InstitutionService.shared.institutionUserSummary(
+          {
+            page: page,
```
Comment (henryoforeh-aot) [inline sources/packages/web/src/components/common/InstitutionUserSummary.vue:219]: Why so much space here ?

File: sources/packages/api/src/route-controllers/institution/institution.aest.controller.ts (hunk 1 above)
Comment (ann-aot) [reply sources/packages/api/src/route-controllers/institution/institution.aest.controller.ts:123]: maybe you can also add a comment

File: sources/packages/web/src/services/http/Institution.ts [hunk 15]
```diff
@@ -1,97 +1,73 @@
...
   }
 
   public async updateInstitution(
-    data: InstitutionContactDTO | InstitutionProfileDTO,
+    data: InstitutionContactAPIInDTO | InstitutionProfileAPIInDTO,
     institutionId?: number,
   ): Promise<void> {
     const url = institutionId ? `institution/${institutionId}` : "institution";
-    await this.patchCall<InstitutionContactDTO>(this.addClientRoot(url), data);
+    await this.patchCall<
```
Comment (henryoforeh-aot) [inline sources/packages/web/src/services/http/Institution.ts:38]: A lot of space here.

File: sources/packages/api/src/services/institution/institution.service.ts [hunk 16]
```diff
@@ -17,25 +17,20 @@ import {
...
-import {
-  sortUsersColumnMap,
-  FieldSortOrder,
-  DEFAULT_PAGE_NUMBER,
-  DEFAULT_PAGE_LIMIT,
-} from "../../utilities";
+import { sortUsersColumnMap, PaginationOptions } from "../../utilities";
 import { InstitutionUserRoles } from "../../auth/user-types.enum";
-import { UpdateInstitution } from "./institution.service.model";
+import {
```
Comment (henryoforeh-aot) [inline sources/packages/api/src/services/institution/institution.service.ts:27]: Please remove extra space here

File: sources/packages/api/src/route-controllers/institution/institution.aest.controller.ts [hunk 17]
```diff
@@ -64,4 +79,96 @@ export class InstitutionAESTController extends BaseController {
...
+        country: eachInstitution.institutionAddress.country,
+        postalCode: eachInstitution.institutionAddress.postalCode,
+      },
+    }));
+  }
+
+  /**
+   * Get the Basic Institution info for the ministry institution detail page.
+   * @param institutionId
+   * @returns BasicInstitutionInfo
```
Comment (ann-aot) [inline sources/packages/api/src/route-controllers/institution/institution.aest.controller.ts:155]: please update the return comment

**Review by henryoforeh-aot — APPROVED** [review]

File: sources/packages/api/src/route-controllers/institution/institution.controller.service.ts [hunk 18]
```diff
@@ -75,20 +68,26 @@ export class InstitutionControllerService {
   }
 
   /**
-   * Validate dry run submission.
-   * @param payload form data to validate.
+   * Get institution users with page,sort and search.
```
Comment (ann-aot) [inline sources/packages/api/src/route-controllers/institution/institution.controller.service.ts:71]: space after `,` in `page,sort`

File: sources/packages/api/src/route-controllers/institution/institution.aest.controller.ts (hunk 1 above)
Comment (dheepak-aot) [reply sources/packages/api/src/route-controllers/institution/institution.aest.controller.ts:123]: Thanks for letting know @ann-aot. Will do that.

File: sources/packages/api/src/route-controllers/institution/institution.aest.controller.ts (hunk 11 above)
Comment (dheepak-aot) [reply sources/packages/api/src/route-controllers/institution/institution.aest.controller.ts:108]: I am changing.

File: sources/packages/api/src/route-controllers/institution/institution.institutions.controller.ts [hunk 19]
```diff
@@ -27,10 +71,77 @@ export class InstitutionInstitutionsController extends BaseController {
...
+    private readonly bceidAccountService: BCeIDService,
+    private readonly locationService: InstitutionLocationService,
   ) {
     super();
   }
 
+  /**
+   * Create institution on Institution setup process.
+   * @param payload
+   * @param userToken
```
Comment (ann-aot) [inline sources/packages/api/src/route-controllers/institution/institution.institutions.controller.ts:84]: return comment missing

File: sources/packages/api/src/route-controllers/institution/institution.institutions.controller.ts [hunk 20]
```diff
@@ -27,10 +71,77 @@ export class InstitutionInstitutionsController extends BaseController {
...
+    };
+  }
+
+  /**
+   * Controller method to get all institution users for institution Admin.
+   * @queryParm page, page number if nothing is passed then
+   * DEFAULT_PAGE_NUMBER is taken
+   * @queryParm pageLimit, page size or records per page, if nothing is
+   * passed then DEFAULT_PAGE_LIMIT is taken
+   * @queryParm searchName, user's name keyword to be searched
```
Comment (ann-aot) [inline sources/packages/api/src/route-controllers/institution/institution.institutions.controller.ts:117]: its `searchCriteria` not `searchName`

File: sources/packages/api/src/route-controllers/institution/institution.institutions.controller.ts [hunk 21]
```diff
@@ -39,7 +150,7 @@ export class InstitutionInstitutionsController extends BaseController {
   @Get()
   async getInstitutionDetail(
     @UserToken() token: IInstitutionUserToken,
-  ): Promise<InstitutionDetailDTO> {
+  ): Promise<InstitutionDetailAPIOutDTO> {
```
Comment (ann-aot) [inline sources/packages/api/src/route-controllers/institution/institution.institutions.controller.ts:153]: please update the return comment

File: sources/packages/web/src/components/common/InstitutionUserSummary.vue (hunk 14 above)
Comment (dheepak-aot) [reply sources/packages/web/src/components/common/InstitutionUserSummary.vue:219]: It is formatted by prettier. What is the issue here @henryoforeh-aot ?

image

File: sources/packages/api/src/route-controllers/institution/institution.institutions.controller.ts [hunk 22]
```diff
@@ -52,12 +163,356 @@ export class InstitutionInstitutionsController extends BaseController {
...
+  @IsInstitutionAdmin()
+  @Get("user-types-roles")
+  async getUserTypesAndRoles(): Promise<InstitutionUserTypeAndRoleAPIOutDTO> {
+    return this.institutionService.getUserTypesAndRoles();
+  }
+
+  /**
+   * Get institution user by user name(guid).
+   * @param userName
+   * @returns
```
Comment (ann-aot) [inline sources/packages/api/src/route-controllers/institution/institution.institutions.controller.ts:258]: please update the return comment

File: sources/packages/api/src/route-controllers/institution/models/institution-user.dto.ts [hunk 23]
```diff
@@ -0,0 +1,121 @@
...
+  authorizations: InstitutionLocationAuthAPIOutDTO[];
+}
+
+export class InstitutionUserDetailAPIOutDTO {
+  id?: number;
+  user: InstitutionUserSummaryAPIOutDTO;
+  authorizations: UserAuthDetailAPIOutDTO;
+}
+
+export interface InstitutionUserLocationsAPIOutDTO {
```
Comment (ann-aot) [inline sources/packages/api/src/route-controllers/institution/models/institution-user.dto.ts:104]: shouldn't it be class?

File: sources/packages/api/src/route-controllers/institution/models/institution.dto.ts [hunk 24]
```diff
@@ -28,7 +28,7 @@ export class CreateInstitutionDto {
   website: string;
   @IsOptional()
   regulatingBody: string;
-  @IsNotEmpty()
+  @IsDate()
   establishedDate: Date;
```
Comment (ann-aot) [inline sources/packages/api/src/route-controllers/institution/models/institution.dto.ts:32]: are you sure if `@IsDate()` works? I think formio return the date string

File: sources/packages/api/src/route-controllers/institution/models/institution.dto.ts [hunk 25]
```diff
@@ -28,7 +28,7 @@ export class CreateInstitutionDto {
   website: string;
   @IsOptional()
   regulatingBody: string;
-  @IsNotEmpty()
+  @IsDate()
```
Comment (ann-aot) [inline sources/packages/api/src/route-controllers/institution/models/institution.dto.ts:31]: are you sure if `@IsDate()` works? I think formio returns the date string

File: sources/packages/api/src/route-controllers/institution/models/institution.dto.ts [hunk 26]
```diff
@@ -124,9 +142,9 @@ export interface InstitutionDetailDto {
   isBCPrivate?: boolean;
 }
 
-export interface SearchInstitutionRespDto {
+export interface SearchInstitutionAPIOutDTO {
```
Comment (ann-aot) [inline sources/packages/api/src/route-controllers/institution/models/institution.dto.ts:145]: class?

File: sources/packages/api/src/route-controllers/models/common.dto.ts [hunk 27]
```diff
@@ -46,7 +46,7 @@ export class AddressInfoInDTO {
  * Common DTO for Address.
  ** This DTO is returned as API response body.
  */
-export class AddressOutDTO {
+export class AddressAPIOutDTO {
   addressLine1: string;
   addressLine2: string;
```
Comment (ann-aot) [inline sources/packages/api/src/route-controllers/models/common.dto.ts:51]: addressLine2 is optional right? so `addressLine2?`

File: sources/packages/api/src/services/bceid/search-bceid.model.ts [hunk 28]
```diff
@@ -35,8 +35,8 @@ export class SearchAccountOptions {
  * Pagination options allowed on BCeID Web Service search request.
  */
 export class PaginationOptions {
-  pageSize?: Number = PAGINATION_MAX_PAGE_SIZE;
```
Comment (ann-aot) [inline sources/packages/api/src/services/bceid/search-bceid.model.ts:38]: 👍

File: sources/packages/api/src/services/institution/institution.service.model.ts [hunk 29]
```diff
@@ -20,3 +20,44 @@ export interface UpdateInstitution {
...
+  primaryEmail: string;
+  website: string;
+  regulatingBody: string;
+  establishedDate: Date;
+  primaryContactFirstName: string;
+  primaryContactLastName: string;
+  primaryContactEmail: string;
+  primaryContactPhone: string;
+  addressLine1: string;
+  addressLine2: string;
```
Comment (ann-aot) [inline sources/packages/api/src/services/institution/institution.service.model.ts:37]: `addressLine2?`?

File: sources/packages/api/src/services/institution/institution.service.ts [hunk 30]
```diff
@@ -288,13 +283,9 @@ export class InstitutionService extends RecordDataModelService<Institution> {
...
    */
-  async allUsers(
-    searchName: string,
-    sortField: string,
+  async getInstitutionUsers(
     institutionId: number,
-    page = DEFAULT_PAGE_NUMBER,
-    pageLimit = DEFAULT_PAGE_LIMIT,
-    sortOrder = FieldSortOrder.ASC,
+    paginationOptions: PaginationOptions,
```
Comment (ann-aot) [inline sources/packages/api/src/services/institution/institution.service.ts:288]: please update the comments too

File: sources/packages/web/src/services/InstitutionService.ts [hunk 31]
```diff
@@ -172,45 +171,45 @@ export class InstitutionService {
...
    * @returns All the institution users.
    */
-  public async institutionSummary(
-    page = DEFAULT_PAGE_NUMBER,
-    pageCount = DEFAULT_PAGE_LIMIT,
-    searchName?: string,
-    sortField?: UserFields,
-    sortOrder?: DataTableSortOrder,
+  public async institutionUserSummary(
+    paginationOptions: PaginationOptions,
```
Comment (ann-aot) [inline sources/packages/web/src/services/InstitutionService.ts:175]: please update the comment too

File: sources/packages/web/src/services/InstitutionService.ts [hunk 32]
```diff
@@ -368,50 +369,10 @@ export class InstitutionService {
    */
   async getBasicInstitutionInfoById(
     institutionId: number,
-  ): Promise<BasicInstitutionInfo> {
+  ): Promise<InstitutionBasicAPIOutDTO> {
```
Comment (ann-aot) [inline sources/packages/web/src/services/InstitutionService.ts:372]: please update the comment

File: sources/packages/web/src/services/http/dto/Common.dto.ts [hunk 33]
```diff
@@ -2,3 +2,66 @@ export interface OptionItemAPIOutDTO {
...
+ * TODO: Address DTOs have been with 2 different names as the
+ * TODO: property province is used as provinceState in some places in code.
+ **/
+/**
+ * Common DTO for Address
+ ** This DTO is used in API request body.
+ */
+export interface AddressAPIInDTO {
+  addressLine1: string;
+  addressLine2: string;
```
Comment (ann-aot) [inline sources/packages/web/src/services/http/dto/Common.dto.ts:16]: `addressLine2?`?

File: sources/packages/web/src/services/http/dto/Common.dto.ts [hunk 34]
```diff
@@ -2,3 +2,66 @@ export interface OptionItemAPIOutDTO {
...
+  postalCode: string;
+}
+
+/**
+ * Common DTO for Address
+ ** This DTO is used in API request body.
+ */
+export interface AddressInfoAPIInDTO {
+  addressLine1: string;
+  addressLine2: string;
```
Comment (ann-aot) [inline sources/packages/web/src/services/http/dto/Common.dto.ts:29]: `addressLine2?`?

File: sources/packages/web/src/services/http/dto/Common.dto.ts [hunk 35]
```diff
@@ -2,3 +2,66 @@ export interface OptionItemAPIOutDTO {
...
+  postalCode: string;
+}
+
+/**
+ * Common DTO for Address.
+ ** This DTO is returned as API response body.
+ */
+export interface AddressAPIOutDTO {
+  addressLine1: string;
+  addressLine2: string;
```
Comment (ann-aot) [inline sources/packages/web/src/services/http/dto/Common.dto.ts:42]: `addressLine2?`?

File: sources/packages/web/src/services/http/dto/Institution.dto.ts [hunk 36]
```diff
@@ -0,0 +1,176 @@
...
+  primaryEmail: string;
+  website: string;
+  regulatingBody: string;
+  establishedDate: Date;
+  primaryContactFirstName: string;
+  primaryContactLastName: string;
+  primaryContactEmail: string;
+  primaryContactPhone: string;
+  addressLine1: string;
+  addressLine2: string;
```
Comment (ann-aot) [inline sources/packages/web/src/services/http/dto/Institution.dto.ts:122]: `addressLine2?`?

File: sources/packages/web/src/services/http/Institution.ts (hunk 15 above)
Comment (dheepak-aot) [reply sources/packages/web/src/services/http/Institution.ts:38]: Again added by prettier.

File: sources/packages/api/src/route-controllers/institution/models/institution-user.dto.ts (hunk 23 above)
Comment (dheepak-aot) [reply sources/packages/api/src/route-controllers/institution/models/institution-user.dto.ts:104]: Yes it is. Changed.

File: sources/packages/api/src/route-controllers/institution/models/institution.dto.ts (hunk 25 above)
Comment (dheepak-aot) [reply sources/packages/api/src/route-controllers/institution/models/institution.dto.ts:31]: Yes @IsDate works well with date string.

**Review by guru-aot — APPROVED** [review]
LGTM, thanks for doing the reorganizing. Also we should look on the test cases in a separate story or should remove it, if its not needed.

File: sources/packages/api/src/route-controllers/institution/models/institution.dto.ts (hunk 25 above)
Comment (dheepak-aot) [reply sources/packages/api/src/route-controllers/institution/models/institution.dto.ts:31]: Here is an example of same scenario with @IsDate() class validator. 

image

image

Comment (ann-aot) [reply sources/packages/api/src/route-controllers/institution/models/institution.dto.ts:31]: Thanks @dheepak-aot for the details

**Review by ann-aot — APPROVED** [review]
Nice work @dheepak-aot . Thanks for doing the changes 👍
